### PR TITLE
Feature backend steps and templates 28

### LINKED ISSUE #28: BE: Steps Model + Email Templates
- Each campaign has many steps
- One step has one email template
      - an email template: subject, and a body (HTML)

- POST /campaigns/:id/steps -> create a new step for a campaign
     - you will expect the details about an email template, automatically create a template in the back-end and associated it with this step
      - modify the get one campaign by id route to include all the steps with email templates
- PUT /campaigns/:id/steps/:id -> edit a step in a campaign

### PR DESCRIPTION
**Pull request by christianromarserad**
Changes:
- new routes for creating and updating campaign step
- new step model and migration script

Routes:
- POST "/api/campaign/:id/step"   - creating a step for the campaign
- PUT "/api/campaign/:id/step/:order"   - Updating a step

### CODE REVIEW

File: server/src/controllers/campaign.js [hunk 1]
```diff
@@ -80,6 +80,63 @@ class CampaignController {
...
+        userId: req.currentUser.id
+      },
+      include: {
+        model: db.Step,
+        required: false,
+        where: {
+          order: req.params.order
+        }
+      }
+    });
```
Comment (skassam21) [inline server/src/controllers/campaign.js:119]: This should be a simple get by order and compaign id - it's more efficient database call then this where statement

File: server/src/controllers/campaign.js [hunk 2]
```diff
@@ -80,6 +80,63 @@ class CampaignController {
...
   };
+
+  static addStep = async (req, res) => {
+    const campaign = await db.Campaign.findOne({
+      where: {
+        id: req.params.id,
+        userId: req.currentUser.id
+      },
+      include: db.Step
+    });
```
Comment (skassam21) [inline server/src/controllers/campaign.js:92]: More efficient to do a Select COUNT SQL statement - do you know if Sequelize can do that?

**Review by skassam21 — COMMENTED** [review]
You can make more efficient API calls

File: server/src/controllers/campaign.js (hunk 1 above)
Comment (christianromarserad) [reply server/src/controllers/campaign.js:119]: Updated to get the step using campaign id and order

File: server/src/controllers/campaign.js (hunk 2 above)
Comment (christianromarserad) [reply server/src/controllers/campaign.js:92]: There is a Sequelize.fn but I couldn't get it to work with an association.

**Pull request merged**